xray/segment.go

```diff
@@ -126,9 +139,28 @@ func BeginSegmentWithSampling(ctx context.Context, name string, r *http.Request,
 		seg.Dummy = true
 	}
 
+	// generates segment and trace id based on sampling decision and secure random env variable
+	idGeneration(seg)
+
 	return context.WithValue(ctx, ContextKey, seg), seg
 }
 
+func idGeneration(seg *Segment) {
+	secureRandom := os.Getenv("AWS_XRAY_SECURE_RANDOM_ID")
```

Comment:
Not sure how I feel about this env var name, especially since the secure random vs. fast random configuration, which is a different configuration than this, may be added in the future. How about `AWS_XRAY_UNSAMPLED_IDS`? Or `AWS_XRAY_NOOP_ID` (which is true by default, so the opposite conditions as we have currently)?

Reply:
Yeah, I like `AWS_XRAY_NOOP_ID` name and setting variable as false should generate secure random IDs even for un-sampled traces.